src/test/java/solver/script/solvable_modification/SolvableModificationVisitorTest.java

```diff
@@ -54,11 +130,51 @@ private SolvableModificationId mockSampleIds() {
 		return sampleId;
 	}
 
-	private BinaryExpressionDelegate createMockExpressionToSolve() {
+	private BinaryExpressionDelegate createMockBinaryExpressionToSolve() {
 		BinaryExpressionDelegate sampleInput = mock(BinaryExpressionDelegate.class);
 		TypeDelegate mockTypeDelegate = mock(TypeDelegate.class);
 		when(mockTypeDelegate.asEnum()).thenReturn(Type.INT);
 		when(sampleInput.getType()).thenReturn(mockTypeDelegate);
 		return sampleInput;
 	}
+
+	private UnaryExpressionDelegate createMockUnaryExpressionToSolve() {
+		UnaryExpressionDelegate sampleInput = mock(UnaryExpressionDelegate.class);
+		TypeDelegate mockTypeDelegate = mock(TypeDelegate.class);
+		when(mockTypeDelegate.asEnum()).thenReturn(Type.INT);
+		when(sampleInput.getType()).thenReturn(mockTypeDelegate);
+		return sampleInput;
+	}
+
+	private AssignExpressionDelegate createMockAssignExpressionToSolve() {
+		AssignExpressionDelegate sampleInput = mock(AssignExpressionDelegate.class);
+		TypeDelegate mockTypeDelegate = mock(TypeDelegate.class);
+		when(mockTypeDelegate.asEnum()).thenReturn(Type.INT);
+		when(sampleInput.getType()).thenReturn(mockTypeDelegate);
```

Comment:
We can move these 3 statements to their own function for better code reuse as all these 'createMockxyz' functions do this.